packages/core-utils/src/itinerary.ts

```diff
@@ -566,7 +566,7 @@ export function getLegCost(
   leg: Leg,
   mediumId: string | null,
   riderCategoryId: string | null
-): { price?: Money; transferAmount?: Money | undefined } {
+): { price?: Money; transferAmount?: Money | undefined; useId?: string } {
```

Comment:
Maybe rename `useId` to `productId`?

Reply:
The thing is, it's _not_ the product id. It is an id that refers to the "instance" of the product, so for example a day pass that you use several times, hence "useId".

The product id globally identifies the product that you're buying and probably has an id like `atl:daypass` but the "useId" is an identifier local to the itinerary allows you to deduplicate the day pass so you know you don't have to buy three passes but it's applicable to all of the legs.

Reply:
Okay. How about `ticketId`?

Reply:
How about `productUseId` to more closely match what it's called in the backend?

Reply:
Sure